lab_3_generate_by_ngrams/main.py

```diff
@@ -221,6 +328,13 @@ def generate_next_token(self, sequence: tuple[int, ...]) -> Optional[dict]:
 
         In case of corrupt input arguments, None is returned
         """
+        if not (isinstance(sequence, tuple) and sequence
+                and len(sequence) >= self._n_gram_size - 1):
+            return None
+        context = sequence[-self._n_gram_size + 1:]
+        return {n_gram[-1]: freq
```

Comment:
use `dict(filter(...))`

Reply:
Well, I've tried, but it still needed dictionary comprehension🥲